workers/unity/Packages/io.improbable.gdk.core/.codegen/Source/Generators/Core/CommandDiffStorageGenerator.cs

```diff
@@ -0,0 +1,225 @@
+
+    private bool requestsSorted;
+    private bool responsesSorted;
+
+    public uint GetComponentId()
+    {{
+        return ComponentId;
+    }}
+
+    public uint GetCommandId()
+    {{
+        return {command.CommandIndex};
+    }}
+
+    public Type GetRequestType()
+    {{
+        return typeof({receivedRequestType});
+    }}
+
+    public Type GetResponseType()
+    {{
+        return typeof({receivedResponseType});
+    }}
+
+    public void Clear()
+    {{
+        requestStorage.Clear();
+        responseStorage.Clear();
+        requestsSorted = false;
+        responsesSorted = false;
+    }}
+
+    public void RemoveRequests(long entityId)
+    {{
+        requestStorage.RemoveAll(request => request.EntityId.Id == entityId);
+    }}
+
+    public void AddRequest({receivedRequestType} request)
+    {{
+        requestStorage.Add(request);
+    }}
+
+    public void AddResponse({receivedResponseType} response)
+    {{
+        responseStorage.Add(response);
+    }}
+
+    public MessagesSpan<{receivedRequestType}> GetRequests()
+    {{
+        return requestStorage.Slice();
+    }}
+
+    public MessagesSpan<{receivedRequestType}> GetRequests(EntityId targetEntityId)
+    {{
+        if (!requestsSorted)
+        {{
+            requestStorage.Sort(requestComparer);
+            requestsSorted = true;
+        }}
+
+        var (firstIndex, count) = requestStorage.GetEntityRange(targetEntityId);
+        return requestStorage.Slice(firstIndex, count);
+    }}
+
+    public MessagesSpan<{receivedResponseType}> GetResponses()
+    {{
+        return responseStorage.Slice();
+    }}
+
+    public MessagesSpan<{receivedResponseType}> GetResponse(long requestId)
+    {{
+        if (!responsesSorted)
+        {{
+            responseStorage.Sort(responseComparer);
+            responsesSorted = true;
+        }}
+
+        var responseIndex = responseStorage.GetResponseIndex(requestId);
+        return responseIndex.HasValue
+            ? responseStorage.Slice(responseIndex.Value, 1)
+            : MessagesSpan<{receivedResponseType}>.Empty();
+    }}
+
+    private class RequestComparer : IComparer<{receivedRequestType}>
+    {{
+        public int Compare({receivedRequestType} x, {receivedRequestType} y)
+        {{
+            return x.EntityId.Id.CompareTo(y.EntityId.Id);
+        }}
+    }}
+
+    private class ResponseComparer : IComparer<{receivedResponseType}>
+    {{
+        public int Compare({receivedResponseType} x, {receivedResponseType} y)
+        {{
+            return x.RequestId.CompareTo(y.RequestId);
+        }}
+    }}
+}}
+");
+        }
+
+        private static Text GenerateCommandsToSendStorage(UnityCommandDetails command, string qualifiedNamespace, string componentName)
+        {
+            Logger.Trace($"Generating {qualifiedNamespace}.{componentName}.{command.CommandName}CommandsToSendStorage class.");
+
+            var requestType = $"{command.CommandName}.Request";
```

Comment:
These can also go?

Reply:
I've left it in where we use it several times